checker/src/main/java/org/checkerframework/checker/calledmethods/builder/LombokSupport.java

```diff
@@ -0,0 +1,204 @@
+        List<String> defaultedPropertyNames = new ArrayList<>();
+        for (Element member : lombokClassElement.getEnclosedElements()) {
+            if (member.getKind() == ElementKind.FIELD) {
+                for (AnnotationMirror anm :
+                        atypeFactory.getElementUtils().getAllAnnotationMirrors(member)) {
+                    if (NONNULL_ANNOTATIONS.contains(AnnotationUtils.annotationName(anm))) {
+                        requiredPropertyNames.add(member.getSimpleName().toString());
+                    }
+                }
+            } else if (member.getKind() == ElementKind.METHOD
+                    && ElementUtils.hasAnnotation(member, "lombok.Generated")) {
+                String methodName = member.getSimpleName().toString();
+                // Handle fields with @Builder.Default annotations.
+                // If a field foo has an @Builder.Default annotation, Lombok always generates a
+                // method called $default$foo.
+                if (methodName.startsWith("$default$")) {
+                    String propName = methodName.substring(9); // $default$ has 9 characters
+                    defaultedPropertyNames.add(propName);
+                }
+            } else if (member.getKind().isClass() && member.toString().endsWith("Builder")) {
+                // If a field bar has an @Singular annotation, Lombok always generates a method
+                // called clearBar in the builder class itself. Therefore, search the builder for
+                // such a method, and extract the appropriate property name to treat as defaulted.
+                for (Element builderMember : member.getEnclosedElements()) {
+                    if (builderMember.getKind() == ElementKind.METHOD
+                            && ElementUtils.hasAnnotation(builderMember, "lombok.Generated")) {
+                        String methodName = builderMember.getSimpleName().toString();
+                        if (methodName.startsWith("clear")) {
+                            String propName =
+                                    Introspector.decapitalize(
+                                            methodName.substring(5)); // clear has 5 characters
+                            defaultedPropertyNames.add(propName);
+                        }
+                    } else if (builderMember.getKind() == ElementKind.FIELD) {
+                        VariableTree variableTree =
+                                (VariableTree) atypeFactory.declarationFromElement(builderMember);
+                        if (variableTree != null && variableTree.getInitializer() != null) {
+                            String propName = variableTree.getName().toString();
+                            defaultedPropertyNames.add(propName);
+                            defaultedElements.put(builderMember, propName);
+                        } else if (defaultedElements.containsKey(builderMember)) {
```

Comment:
I don't think this code relies on the order of elements, but even if it does it's not a concern because Lombok will always generate things in the same order (and it's not a soundness concern, so we'd get FP warnings that would indicate the problem if the generator changes).